src/lib/storage/table.cpp

```diff
@@ -17,53 +17,63 @@
 
 namespace opossum {
 
+Table::Table(const uint32_t chunk_size) {
+  _chunk_size = chunk_size;
+  _add_chunk();
+}
+
 void Table::add_column(const std::string& name, const std::string& type) {
-  // Implementation goes here
+  _column_names.push_back(name);
+  _column_types.push_back(type);
+  _chunks.back()->add_segment(make_shared_by_data_type<BaseSegment, ValueSegment>(type));
 }
 
 void Table::append(std::vector<AllTypeVariant> values) {
-  // Implementation goes here
+  if (_chunks.back()->size() == _chunk_size) {
+    _add_chunk();
+  }
+  _chunks.back()->append(values);
 }
 
-uint16_t Table::column_count() const {
-  // Implementation goes here
-  return 0;
-}
+uint16_t Table::column_count() const { return _column_types.size(); }
 
 uint64_t Table::row_count() const {
-  // Implementation goes here
-  return 0;
+  uint64_t count = 0;
+  for (auto const chunk : _chunks) {
+    count += chunk->size();
+  }
+  return count;
 }
 
-ChunkID Table::chunk_count() const {
-  // Implementation goes here
-  return ChunkID{0};
-}
+ChunkID Table::chunk_count() const { return static_cast<ChunkID>(_chunks.size()); }
 
 ColumnID Table::column_id_by_name(const std::string& column_name) const {
-  // Implementation goes here
-  return ColumnID{0};
+  auto const pos = std::find(_column_names.begin(), _column_names.end(), column_name);
+  if (pos == _column_names.end()) {
+    throw std::runtime_error("Implement Table::get_chunk");
```

Comment:
What's with this one?

Reply:
Aye, good call, didn't change the exception message.